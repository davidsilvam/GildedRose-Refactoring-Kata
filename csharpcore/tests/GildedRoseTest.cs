using Xunit;
using System.Collections.Generic;
using Xunit.Abstractions;//added to debug

using ApprovalTests;
using ApprovalTests.Reporters;

using System.Text;

namespace csharpcore
{
    [UseReporter(typeof(DiffReporter))]
    public class GildedRoseTest
    {
        private readonly ITestOutputHelper _output;

        public GildedRoseTest(ITestOutputHelper output)
        {
            _output = output;
        }

        [Fact]
        public void foo()
        {
            IList<Item> Items = new List<Item> { new Item { Name = "foo", SellIn = 0, Quality = 0 } };
            GildedRose app = new GildedRose(Items);
            app.UpdateQuality();
            //  _output.WriteLine("Debug");
            Assert.Equal("foo", Items[0].Name);
        }

        [Fact]
        public void NormalItemIncreaseAndDecrease()
        {
            IList<Item> Items = new List<Item> {
                new Item {
                    Name = "+5 Dexterity Vest",
                    SellIn = 10,
                    Quality = 20
                },
                new Item {
                    Name = "Elixir of the Mongoose",
                    SellIn = 5,
                    Quality = 7
                }
            };

            GildedRose app = new GildedRose(Items);

            var fakeoutput = new StringBuilder();

            for (var i = 0; i < 31; i++)
            {
                fakeoutput.Append("-------- day " + i + " --------" + "\n");
                for (var j = 0; j < Items.Count; j++)
                {
                    fakeoutput.Append(Items[j].Name + ", " + Items[j].SellIn + ", " + Items[j].Quality + "\n");
                }
                app.UpdateQuality();
            }

            Approvals.Verify(fakeoutput);
        }

        [Fact]
        public void AgedBrieItemIncreaseAndDecrease()
        {
            IList<Item> Items = new List<Item> {
                new Item {
                    Name = "Aged Brie",
                    SellIn = 2,
                    Quality = 0
                }
            };

            GildedRose app = new GildedRose(Items);

            var fakeoutput = new StringBuilder();

            for (var i = 0; i < 31; i++)
            {
                fakeoutput.Append("-------- day " + i + " --------" + "\n");
                for (var j = 0; j < Items.Count; j++)
                {
                    fakeoutput.Append(Items[j].Name + ", " + Items[j].SellIn + ", " + Items[j].Quality + "\n");
                }
                app.UpdateQuality();
            }

            Approvals.Verify(fakeoutput);
        }

        [Fact]
        public void SulfurasItemIncreaseAndDecrease()
        {
            IList<Item> Items = new List<Item> {
                new Item {
                    Name = "Sulfuras, Hand of Ragnaros",
                    SellIn = 0,
                    Quality = 80
                },
                new Item {
                    Name = "Sulfuras, Hand of Ragnaros",
                    SellIn = -1,
                    Quality = 80
                }
            };

            GildedRose app = new GildedRose(Items);

            var fakeoutput = new StringBuilder();

            for (var i = 0; i < 31; i++)
            {
                fakeoutput.Append("-------- day " + i + " --------" + "\n");
                for (var j = 0; j < Items.Count; j++)
                {
                    fakeoutput.Append(Items[j].Name + ", " + Items[j].SellIn + ", " + Items[j].Quality + "\n");
                }
                app.UpdateQuality();
            }

            Approvals.Verify(fakeoutput);
        }

        [Fact]
        public void BackstagePassesItemIncreaseAndDecrease()
        {
            IList<Item> Items = new List<Item> {
               new Item
                {
                    Name = "Backstage passes to a TAFKAL80ETC concert",
                    SellIn = 15,
                    Quality = 20
                },
                new Item
                {
                    Name = "Backstage passes to a TAFKAL80ETC concert",
                    SellIn = 10,
                    Quality = 49
                },
                new Item
                {
                    Name = "Backstage passes to a TAFKAL80ETC concert",
                    SellIn = 5,
                    Quality = 49
                }
            };

            GildedRose app = new GildedRose(Items);

            var fakeoutput = new StringBuilder();

            for (var i = 0; i < 31; i++)
            {
                fakeoutput.Append("-------- day " + i + " --------" + "\n");
                for (var j = 0; j < Items.Count; j++)
                {
                    fakeoutput.Append(Items[j].Name + ", " + Items[j].SellIn + ", " + Items[j].Quality + "\n");
                }
                app.UpdateQuality();
            }

            Approvals.Verify(fakeoutput);
        }

        [Fact]
        public void ConjuredItemIncreaseAndDecrease()
        {
            IList<Item> Items = new List<Item> {
               new Item {
                   Name = "Conjured Mana Cake",
                   SellIn = 3,
                   Quality = 6
                }
            };

            GildedRose app = new GildedRose(Items);

            var fakeoutput = new StringBuilder();

            for (var i = 0; i < 31; i++)
            {
                fakeoutput.Append("-------- day " + i + " --------" + "\n");
                for (var j = 0; j < Items.Count; j++)
                {
                    fakeoutput.Append(Items[j].Name + ", " + Items[j].SellIn + ", " + Items[j].Quality + "\n");
                }
                app.UpdateQuality();
            }

            Approvals.Verify(fakeoutput);
        }
    }
}