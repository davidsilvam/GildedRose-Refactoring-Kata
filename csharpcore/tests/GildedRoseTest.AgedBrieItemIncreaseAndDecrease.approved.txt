-------- day 0 --------
Aged Brie, 2, 0
-------- day 1 --------
Aged Brie, 1, 1
-------- day 2 --------
Aged Brie, 0, 2
-------- day 3 --------
Aged Brie, -1, 4
-------- day 4 --------
Aged Brie, -2, 6
-------- day 5 --------
Aged Brie, -3, 8
-------- day 6 --------
Aged Brie, -4, 10
-------- day 7 --------
Aged Brie, -5, 12
-------- day 8 --------
Aged Brie, -6, 14
-------- day 9 --------
Aged Brie, -7, 16
-------- day 10 --------
Aged Brie, -8, 18
-------- day 11 --------
Aged Brie, -9, 20
-------- day 12 --------
Aged Brie, -10, 22
-------- day 13 --------
Aged Brie, -11, 24
-------- day 14 --------
Aged Brie, -12, 26
-------- day 15 --------
Aged Brie, -13, 28
-------- day 16 --------
Aged Brie, -14, 30
-------- day 17 --------
Aged Brie, -15, 32
-------- day 18 --------
Aged Brie, -16, 34
-------- day 19 --------
Aged Brie, -17, 36
-------- day 20 --------
Aged Brie, -18, 38
-------- day 21 --------
Aged Brie, -19, 40
-------- day 22 --------
Aged Brie, -20, 42
-------- day 23 --------
Aged Brie, -21, 44
-------- day 24 --------
Aged Brie, -22, 46
-------- day 25 --------
Aged Brie, -23, 48
-------- day 26 --------
Aged Brie, -24, 50
-------- day 27 --------
Aged Brie, -25, 50
-------- day 28 --------
Aged Brie, -26, 50
-------- day 29 --------
Aged Brie, -27, 50
-------- day 30 --------
Aged Brie, -28, 50
